### ISSUE TITLE
feature: prefer-invoke

### ISSUE THREAD

**Issue by nirnaor** [wix/eslint-plugin-lodash#155]
In many places there are usages of _.isFunction before running a certain function. _.invoke does exactly that. The rule will prefer using _.invoke instead of checking if something is a function before executing it

# Examples
## Bad
```javascript
// This is bad
if (_.isFunction(obj.foo)) {
    obj.foo(a, b, c)
}
```

## Good
```javascript
// This is good
_.invoke(obj, 'foo', a, b, c)
```

When using !_isFunction in order to raise an exception, the rule should pass
```javascript
// this is alright
if (!_.isFunction(obj, 'foo') {
   throw new Error('foo is not a function')
```

**Comment by ganimomer** [comment]
@nirnaor looks OK, feel free to open a PR.
Things to make sure:
- The function/method call is the *only* statement in the `if` block
- This should warn both on function calls and method calls (e.g. both `f(a, b, c)` and `x.f(a, b, c)`

**Comment by nirnaor** [comment]
Hey @ganimomer 
Just making sure that I understood your notes:
### First note:
##### Good
```javascript
// This is good.
if(_.isFunction(obj.foo) {
  obj.foo();
  console.log('calling foo')
}
```

### Second note
#### Bad
var f = () => console.log('i am a function')
```javascript
// This is bad.
if(_.isFunction(f)){
  f()
}
```

**Comment by ganimomer** [comment]
Exactly. (Also, you can use these examples in the docs)

**Comment by nirnaor** [comment]
Also
```javascript
// This is bad.
if(shouldRender && _.isFunction(obj.render){
  obj.render()
}
```
// While this is good.
if(shouldRender{
  _.invoke(obj, 'render');
}

**Comment by ganimomer** [comment]
Yeah, I agree. However, the order here matters.
This:
```js
if (_.isFunction(f) && someHeavyCalculationThatTakesAWhile(...someArguments)) {
 f()
}
```
Is not the same as
```js
if (someHeavyCalculationThatTakesAWhile(...someArguments)) {
 _.invoke(f);
}
```
Especially if this code runs a lot.

**Comment by ganimomer** [comment]
Also, make sure to check this __bad__ case as well:
```js
if (shouldRender && !shouldSkip && _.isFunction(f)) {
  f();
}
```
